Release notes to Sprint 66 (`25.01`)

nrichers opened the pull request:
## Internal Notes for Reviewers

This PR includes the release notes to Sprint 66, along with refreshed notebooks, Python API reference, and test descriptions. 

- **Preview: [January 31, 2025](https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-7971/release-notes-to-sprint-66-25-01/releases/2025/2025-jan-31/release-notes.html)** 
- **Source: [release-notes.qmd](https://github.com/validmind/documentation/pull/634/files#diff-779ad6f2e0dcd155eb3f46fc1d02c514394ae2b3ea27960eda697a1c2da8e831)**

## Notes

- This is the first release to follow our new versioning scheme (`25.01`) — I added a note at the top to that effect.
- All admin UI PRs are commented out for lack of detail, except [Configurable default models](https://docs-demo.vm.validmind.ai/pr_previews/nrichers/sc-7971/release-notes-to-sprint-66-25-01/releases/2025/2025-jan-31/release-notes.html#configurable-default-models) — should this be there?

### Other improvements

- @validbeck: PR previews were broken due to demo notebook breaking changes. You can now toggle notebook execution using [ENABLE_DEMO_NOTEBOOK](https://github.com/validmind/documentation/settings/variables/actions/ENABLE_DEMO_NOTEBOOK).

<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `deprecation`
- `documentation`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## External Release Notes

<!--- REPLACE THIS COMMENT WITH YOUR DESCRIPTION --->

validbeck commented:
> PR previews were broken due to demo notebook breaking changes. You can now toggle notebook execution using [ENABLE_DEMO_NOTEBOOK](https://github.com/validmind/documentation/settings/variables/actions/ENABLE_DEMO_NOTEBOOK).

Neat! Will be fixed in PR https://github.com/validmind/documentation/pull/633 and the issue was [reviewed and resolved by John](https://validmind.slack.com/archives/C04U7TKHH62/p1738099399278699) but good to have in the future.

validbeck reviewed (changes requested):
Left some comments/suggestions inline for improvements. Also:

The images in the release folder need to have human-readable names, please. Half the audit effort was me renaming files called `hskjfhjskfg.png` etc. I know this is a minor quibble but this is just a good habit to have.

nrichers commented:
> **EDIT:** Also, the contents of the "Release highlights" section should be encased in the `.highlights` class and use the green buttons:

I feel like this is another area where we've added a layer of complexity to the reading and authoring experience that might best be pruned again? I don't think this buttonology adds anything of value to the reader, and it makes the authoring experience worse. 

From what I recall ...

- we started adding buttons for highlights to make them stand out more;
- next, buttons crept into many of the links throughout the release notes; I objected but then relented
- and now we are differentiating between buttons for highlights and other release note items that have to be coded differently

The authoring experience should be able to focus on the content — the fact that one button is solid dark green and the other a pink outline is a nuance that doesn't really add to the storytelling about the release. 

Your other review comments all make sense and I will address them. That said, I have a concern here about how much time we put into release notes — all of the spit & polish items you suggested make sense, but they take time.

mehdi0501 commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+URL: https://github.com/validmind/validmind-library/pull/256
+Labels: enhancement
+--->
+#### Add context to LLM-based descriptions
```
"Add context to enhance LLM-based text generation for model test results

nrichers replied:
Updated as suggested.

mehdi0501 commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+--->
+#### Add context to LLM-based descriptions
+
+You can now include contextual information in LLM-based descriptions. This enhancement improves test result descriptions by incorporating additional context that can be specified through environment variables.
```
```suggestion
You can now include contextual information to enhance LLM-based generation of test results descriptions and interpretations. This enhancement improves test result descriptions by incorporating additional context that can be specified through environment variables.
```

mehdi0501 commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+:::: {.flex .flex-wrap .justify-around}
+
+::: {.w-50-ns .pr4}
+- **New notebooks**: Two Jupyter Notebooks demonstrate how to document application scorecard models using the {{< var vm.developer >}}. 
```
the difference between the two notebooks is not clear

nrichers replied:
Clarified in the links and text — you can choose from three different approaches: **running individual tests**, **running a full test suite**, or **using a single function** to document a model.

mehdi0501 commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+:::
+
+::: {.w-50-ns .tr}
+[Document an application scorecard model](https://jupyterhub.validmind.ai/hub/user-redirect/lab/tree/code_samples/credit_risk/application_scorecard_executive.ipynb){.button}
```
shouldn't this notebook belong in the prior section about credit scorecard models? This section should be focused on the ongoing monitoring notebook

mehdi0501 commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+
+Several enhancements to the {{< var validmind.developer >}} focus on documenting and testing credit risk scorecard models and ongoing monitoring capabilities:
+
+- **New notebooks**: Two new notebooks for application scorecard models and ongoing monitoring provide step-by-step guides for using the {{< var validmind.developer >}} with credit risk datasets:
```
I think we should focus this section on the notebook to illustrate how to use ValidMind to document an ongoing monitoring report for application scorecard.

nrichers replied:
Reworked the section to be only about ongoing monitoring and moved the notebook as suggested (the monitoring is for the model in that notebook, so I included an inline link in the text to it).

mehdi0501 commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+:::: {.flex .flex-wrap .justify-around}
+
+::: {.w-40-ns .pr4}
+Administrators can now create organizations without models, allowing more flexibility in during onboarding. 
```
I believe this is something only "staff admins" for customer-hosted ValidMind will be able to access (confirm with @cachafla ). If yes, then we should make this clear. Also, I don't believe we are explaining anywhere where to access the admin utility?

mehdi0501 reviewed (commented):
Left some comments throughout. There were a couple of new features/enhancements that were in parallel of sprint 66, that I could not find in the notes (check with @cachafla ), but which warrant a call out. Apologies if we talk about them elsewhere
- the enhanced metadata descriptions in the UI 
- new math equation editor and support for LateX formulas
- export model inventory
- export findings

nrichers commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+
+Risk assessment generation has been enhanced to allow you to provide an optional prompt before starting text generation. This feature lets you guide the output, ensuring that the generated text aligns more closely with your specific requirements.
+
+![Optional prompt for risk assessments](402640824-492758fc-2669-46c6-9c81-198e990b9efe.png){.screenshot}
```
True! I will record an animated GIF and avoid the pirates. 🏴‍☠️

nrichers replied:
Replaced with an animated GIF that shows me entering a prompt, the text being generated, and then accepting those risk assessment notes.

nrichers commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+:::: {.flex .flex-wrap .justify-around}
+
+::: {.w-70-ns}
+The `TestResult` class now has a `doc` property to separate static test documentation from LLM-generated descriptions:
```
Reworked to explain static test descriptions and added link. Thank you — your comment helped me understand this feature.

nrichers commented:
> The images in the release folder need to have human-readable names, please. Half the audit effort was me renaming files called `hskjfhjskfg.png` etc. I know this is a minor quibble but this is just a good habit to have, _especially_ as it is outlined in our style guide.

Renamed as suggested (and removed some images we're not actually using).

nrichers commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+URL: https://github.com/validmind/frontend/pull/1117
+Labels: enhancement
+--->
+### Custom reports and dashboards
```
Removed here but added the tech deep dive blog post to the December release notes in 125412aeb0f99b8729302606ff0dfcab1fad5b66.

validbeck replied:
Somehow you borked the ToC when you added stuff to that page, so I've fixed it and since the button didn't match the rest I actually just replaced it with the listing tile as it's special content: 

| Old | New |
|---|---|
|<img width="1472" alt="Screenshot 2025-01-31 at 3 36 50 PM" src="https://github.com/user-attachments/assets/54848a56-a814-4db1-8417-c5ae903f946f" /> | <img width="1271" alt="Screenshot 2025-01-31 at 3 46 38 PM" src="https://github.com/user-attachments/assets/5b58623a-cbdc-49a2-b84e-2a02d6a4bcc0" />|

nrichers commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+URL: https://github.com/validmind/frontend/pull/1117
+Labels: enhancement
+--->
+### Custom reports and dashboards
```
Yup, added.

validbeck replied:
This wasn't entirely done — for example, this was clearly meant to live under "ValidMind Platform" and the `Enhancements` section wasn't updated.

<img width="1281" alt="Screenshot 2025-01-31 at 3 31 57 PM" src="https://github.com/user-attachments/assets/875823e2-e3bc-4cd5-815b-aa225c3857f6" />

Also, from before I even started here, the convention has been to list the library first, then the platform, so I changed the order of the items.

Just doing a proper read-through of the content now.

nrichers commented on site/releases/2025/2025-jan-31/release-notes.qmd:
```diff
@@ -0,0 +1,520 @@
+URL: https://github.com/validmind/frontend/pull/1096
+Labels: enhancement
+--->
+### Simplified workflow nodes
```
Cropped!

validbeck commented:
A note as I edit: **releases need a summary at the top for the CTA**. Please try to do these as we write releases so the person cleaning up yearly releases doesn't have to? Also, since we surface these on our index landing page, it needs the summary to incite people to read.

validbeck reviewed (approved):
In the interest of time I've cleaned things up and made the edits, but I did catch a few broken links/some `.qmd` links out to files that weren't the Python docs as an FYI.

nrichers commented:
@mehdi0501:

> * the enhanced metadata descriptions in the UI

Still investigating ...

> * new math equation editor and support for LateX formulas

This was a fix in Sprint 66 but the related frontend PR was marked as `internal`, meaning we skipped it for the release notes. Docs were added retroactively via https://github.com/validmind/documentation/pull/631 

-> **Added to the release notes via https://github.com/validmind/documentation/pull/634/commits/35b9dc51647a717549c5a80c50cb48321bc18049** 

- https://app.shortcut.com/validmind/story/7976/add-ckeditor5-math-support-deprecate-mathtype
- https://github.com/validmind/frontend/pull/1156

> * export model inventory
> * export findings

Both of these were added in Sprint 64 but the related frontend PR was marked as `internal`, skipped  for the release notes. It doesn't look like we ever flagged the related story as needing documentation, either.

-> tracked via [sc-7539](https://app.shortcut.com/validmind/story/8367/author-missing-docs-for-exporting-model-inventory-and-findings)

- https://app.shortcut.com/validmind/story/7539/initial-support-for-inventory-and-findings-export
- https://github.com/validmind/frontend/pull/1107

@cachafla looks like there's a couple of process breakdowns where 

- user guide updates would have been required but were not flagged
- PRs that should have had release notes were labelled as `internal`

nrichers commented:
> In the interest of time I've cleaned things up and made the edits, but I did catch a few broken links/some `.qmd` links out to files that weren't the Python docs as an FYI.

Thank you for all your editing help, @validbeck! I found a couple more links that needed fixing, but I think these release notes look very clean now. I especially like how you moved some content into the margin. 

(I like to gripe about how much time we spend on release notes but, when you do, the results speak for themselves.)
